custom_components/berlin_transport/config_flow.py

```diff
@@ -0,0 +1,182 @@
+# mypy: disable-error-code="attr-defined,call-arg"
+"""The Berlin (BVG) and Brandenburg (VBB) transport integration."""
+from __future__ import annotations
+
+import logging
+
+from typing import Any, Optional
+import requests
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers import selector
+
+from .const import (
+    API_ENDPOINT,
+    API_MAX_RESULTS,
+    CONF_DEPARTURES_STOP_ID,
+    CONF_DEPARTURES_NAME,
+    CONF_DEPARTURES_DIRECTION,
+    CONF_DEPARTURES_DURATION,
+    CONF_DEPARTURES_WALKING_TIME,
+    CONF_SHOW_API_LINE_COLORS,
+    DOMAIN, # noqa
+)
+
+from .sensor import TRANSPORT_TYPES_SCHEMA
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_SEARCH = "search"
+CONF_FOUND_STOPS = "found_stops"
+CONF_SELECTED_STOP = "selected_stop"
+
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_DEPARTURES_DIRECTION): cv.string,
+        vol.Optional(CONF_DEPARTURES_DURATION): cv.positive_int,
+        vol.Optional(CONF_DEPARTURES_WALKING_TIME, default=1): cv.positive_int,
+        vol.Optional(CONF_SHOW_API_LINE_COLORS, default=False): cv.boolean,
+        **TRANSPORT_TYPES_SCHEMA,
+    }
+)
+
+NAME_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_SEARCH): cv.string,
+    }
+)
+
+# IPv6 is broken, see: https://github.com/public-transport/transport.rest/issues/20
```

Comment:
That's funny 🤣  I think I had these issues without realizing that

Reply:
Took me a bit to figure out why the request took so long in home-assistant while it worked quickly when using the browser. Then I checked with curl and saw it timing out on ipv6 :D